src/analysis/normed_space/normed_group_hom.lean

```diff
@@ -0,0 +1,450 @@
+    (div_le_iff hlt).mp ((real.le_Inf _ bounds_nonempty bounds_bdd_below).2
+    (λ c ⟨_, hc⟩, (div_le_iff hlt).mpr $ by { apply hc })))
+
+theorem le_op_norm_of_le {c : ℝ} {x} (h : ∥x∥ ≤ c) : ∥f x∥ ≤ ∥f∥ * c :=
+le_trans (f.le_op_norm x) (mul_le_mul_of_nonneg_left h f.op_norm_nonneg)
+
+theorem le_of_op_norm_le {c : ℝ} (h : ∥f∥ ≤ c) (x : V₁) : ∥f x∥ ≤ c * ∥x∥ :=
+(f.le_op_norm x).trans (mul_le_mul_of_nonneg_right h (norm_nonneg x))
+
+/-- continuous linear maps are Lipschitz continuous. -/
+theorem lipschitz : lipschitz_with ⟨∥f∥, op_norm_nonneg f⟩ f :=
+lipschitz_with.of_dist_le_mul $ λ x y,
+  by { rw [dist_eq_norm, dist_eq_norm, ←map_sub], apply le_op_norm }
+
+lemma ratio_le_op_norm (x : V₁) : ∥f x∥ / ∥x∥ ≤ ∥f∥ :=
+div_le_of_nonneg_of_le_mul (norm_nonneg _) f.op_norm_nonneg (le_op_norm _ _)
+
+/-- If one controls the norm of every `f x`, then one controls the norm of `f`. -/
+lemma op_norm_le_bound {M : ℝ} (hMp: 0 ≤ M) (hM : ∀ x, ∥f x∥ ≤ M * ∥x∥) :
+  ∥f∥ ≤ M :=
+real.Inf_le _ bounds_bdd_below ⟨hMp, hM⟩
+
+theorem op_norm_le_of_lipschitz {f : normed_group_hom V₁ V₂} {K : ℝ≥0} (hf : lipschitz_with K f) :
+  ∥f∥ ≤ K :=
+f.op_norm_le_bound K.2 $ λ x, by simpa only [dist_zero_right, f.map_zero] using hf.dist_le_mul x 0
+
+/-- If a bounded group homomorphism map is constructed from a group homomorphism via the constructor
+`mk_continuous`, then its norm is bounded by the bound given to the constructor if it is
+nonnegative. -/
+lemma mk_continuous_norm_le (f : V₁ →+ V₂) {C : ℝ} (hC : 0 ≤ C) (h : ∀x, ∥f x∥ ≤ C * ∥x∥) :
+  ∥f.mk_continuous C h∥ ≤ C :=
+op_norm_le_bound _ hC h
+
+/-- If a bounded group homomorphism map is constructed from a group homomorphism via the constructor
+`mk_continuous`, then its norm is bounded by the bound given to the constructor or zero if this
+bound is negative. -/
+lemma mk_continuous_norm_le' (f : V₁ →+ V₂) {C : ℝ} (h : ∀x, ∥f x∥ ≤ C * ∥x∥) :
+  ∥f.mk_continuous C h∥ ≤ max C 0 :=
+op_norm_le_bound _ (le_max_right _ _) $ λ x, (h x).trans $
+  mul_le_mul_of_nonneg_right (le_max_left _ _) (norm_nonneg x)
+
+alias mk_continuous_norm_le ← add_monoid_hom.mk_continuous_norm_le
+alias mk_continuous_norm_le' ← add_monoid_hom.mk_continuous_norm_le'
+
+/-! ### Addition of normed group homs -/
+
+/-- Addition of normed group homs. -/
+instance : has_add (normed_group_hom V₁ V₂) :=
+⟨λ f g, (f.to_add_monoid_hom + g.to_add_monoid_hom).mk_continuous (∥f∥ + ∥g∥) $ λ v, calc
+  ∥f v + g v∥
+      ≤ ∥f v∥ + ∥g v∥ : norm_add_le _ _
+  ... ≤ ∥f∥ * ∥v∥ + ∥g∥ * ∥v∥ : add_le_add (le_op_norm f v) (le_op_norm g v)
+  ... = (∥f∥ + ∥g∥) * ∥v∥ : by rw add_mul⟩
+
+/-- The operator norm satisfies the triangle inequality. -/
+theorem op_norm_add_le : ∥f + g∥ ≤ ∥f∥ + ∥g∥ :=
+mk_continuous_norm_le _ (add_nonneg (op_norm_nonneg _) (op_norm_nonneg _)) _
+
+@[simp] lemma coe_add (f g : normed_group_hom V₁ V₂) : ⇑(f + g) = f + g := rfl
+lemma add_apply (f g : normed_group_hom V₁ V₂) (v : V₁) :
+  ((f + g : normed_group_hom V₁ V₂)) v = f v + g v := rfl
```

Comment:
Oh, the spelling I think I was looking for was
```suggestion
  ⇑(f + g) v = f v + g v := rfl
```

Reply:
Doesn't work for me

Reply:
Odd, thanks for trying